packages/apputils/src/sessioncontext.tsx

```diff
@@ -817,31 +853,7 @@ export class SessionContext implements ISessionContext {
       return false;
     }
 
-    let options: Partial<Kernel.IModel> | undefined;
-    if (preference.id) {
-      options = { id: preference.id };
-    } else {
-      const name = SessionContext.getDefaultKernel({
-        specs: this.specsManager.specs,
-        sessions: this.sessionManager.running(),
-        preference
-      });
-      if (name) {
-        options = { name };
-      }
-    }
-
-    if (options) {
-      try {
-        await this._changeKernel(options);
-        return false;
-      } catch (err) {
-        /* no-op */
-      }
-    }
-
-    // Always fall back to selecting a kernel
-    return true;
+    return this.startKernel();
```

Comment:
_startIfNecessary already checked `preference.shouldStart === false` So _startIfNecessary is backward compatible